app/src/main/java/com/github/tonytangandroid/daggertutorial/MainPresenter.java

```diff
@@ -5,14 +5,17 @@
 public class MainPresenter {
 
   private final GetMessageUseCase getMessageUseCase;
-  private final MainView mainView;
+  private MainView mainView;
 
   @Inject
-  public MainPresenter(MainView mainView, GetMessageUseCase getMessageUseCase) {
-    this.mainView = mainView;
+  public MainPresenter(GetMessageUseCase getMessageUseCase) {
     this.getMessageUseCase = getMessageUseCase;
   }
 
+  public void setMainView(MainView mainView) {
```

Comment:
For the time being, there is no good way to inject `MainView ` through constructor.